generators/client/templates/react/src/main/webapp/app/modules/administration/metrics/metrics-modal.tsx.ejs

```diff
@@ -1,57 +1,175 @@
+    threadDump: {},
+    displayedThreads: [],
+    threadDumpAll: 0,
+    threadDumpBlocked: 0,
+    threadDumpRunnable: 0,
+    threadDumpTimedWaiting: 0,
+    threadDumpWaiting: 0
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.threadDump.threads && nextProps.threadDump !== this.props.threadDump) {
+      this.setState({
+        threadDump: nextProps.threadDump,
+        displayedThreads: nextProps.threadDump.threads
+      }, () => {
+        this.refreshThreadsCounter();
+      });
+    }
+  }
+
+  refreshThreadsCounter = () => {
+    let threadDumpAll = 0;
+    let threadDumpRunnable = 0;
+    let threadDumpWaiting = 0;
+    let threadDumpTimedWaiting = 0;
+    let threadDumpBlocked = 0;
+
+    this.state.threadDump.threads.forEach(t => {
+      switch (t.threadState) {
+        case 'RUNNABLE':
+          threadDumpRunnable++;
+          break;
+        case 'WAITING':
+          threadDumpWaiting++;
+          break;
+        case 'TIMED_WAITING':
+          threadDumpTimedWaiting++;
+          break;
+        case 'BLOCKED':
+          threadDumpBlocked++;
+          break;
+        default:
+          break;
+      }
+    });
+
+    threadDumpAll = threadDumpRunnable + threadDumpWaiting + threadDumpTimedWaiting + threadDumpBlocked;
+    this.setState({ threadDumpAll, threadDumpRunnable, threadDumpWaiting, threadDumpTimedWaiting, threadDumpBlocked });
+  };
+
+  searchFilter = event => {
+    let filteredList = this.state.threadDump.threads;
+    filteredList = filteredList.filter(t => t.lockName != null &&
+                                            t.lockName.toLowerCase().includes(event.target.value.toLowerCase()));
+    this.setState({
+      displayedThreads: filteredList
+    });
+  };
+
+  filterByBadge = badge => {
+    if (badge !== '') {
+      let filteredList = this.state.threadDump.threads;
+      filteredList = filteredList.filter(t => t.threadState === badge);
+      this.setState({
+        displayedThreads: filteredList
+      });
+    } else {
+      this.setState({
+        displayedThreads: this.state.threadDump.threads
+      });
+    }
+  };
+
+  getBadgeClass = threadState => {
+    if (threadState === 'RUNNABLE') {
+      return 'badge-success';
+    } else if (threadState === 'WAITING') {
+      return 'badge-info';
+    } else if (threadState === 'TIMED_WAITING') {
+      return 'badge-warning';
+    } else if (threadState === 'BLOCKED') {
+      return 'badge-danger';
+    }
+  };
+
+  render() {
+    const { showModal, handleClose } = this.props;
+    const { threadDump, displayedThreads } = this.state;
+
+    return <Modal
+      isOpen={showModal} modalTransition={{ timeout: 20 }} backdropTransition={{ timeout: 10 }}
+      toggle={handleClose} className="modal-lg">
+      <ModalHeader toggle={handleClose}>Threads dump</ModalHeader>
+      <ModalBody>
+          <span className="badge badge-primary" onClick={this.filterByBadge.bind(this, '')}>All&nbsp;<span
```

Comment:
no need to bind, just do `onClick={this.filterByBadge('')}`
and change method to return closure
```
filterByBadge = badge => () => {
}
```